tests/purs/failing/SelfImport/Dummy.purs

```diff
@@ -0,0 +1,5 @@
+-- This module only exists so that we perform a full build for the
+-- SelfImport.purs module. If this module didn't exist, we would perform a
+-- single-module fast rebuild, which doesn't perform the `sortModules` step,
+-- and so the error we want to see wouldn't be emitted.
+module Dummy where
```

Comment:
Does this mean that when using fast-build I can do a self import and the compiler will allow it? And the compiler will only give me an error when switching to a full build? I don't think we have any other cases where the compiler conditionally allows something _within_ the module in question.

Reply:
Unfortunately, yes. There are a couple of other examples of this which already do this though, which can be seen by looking inside `Language.PureScript.Make`; specifically, attempting to define a module whose name starts with `Prim`, and giving a module a name which is already taken by some other module. See:

https://github.com/purescript/purescript/blob/b09144bc68637bf6657e4147567a3e9185221594/src/Language/PureScript/Make.hs#L118-L135

Reply:
To me it seems like all of these checks could be done elsewhere. The Prim check can be done for each module, as well as the self import check. And module names could be checked against what purs-ide has in memory (which is as good as it can get, admittedly). Those are all larger changes though, so I'm fine with this for now.